src/app/shared/log-in/log-in.component.ts

```diff
@@ -98,10 +106,22 @@ export class LogInComponent implements OnDestroy, OnInit {
    * Lifecycle hook that is called after data-bound properties of a directive are initialized.
    * @method ngOnInit
    */
-  public ngOnInit() {
-    // set isAuthenticated
+  public ngOnInit() {    // set isAuthenticated
     this.isAuthenticated = this.store.pipe(select(isAuthenticated));
 
+    // for mobile login, set the redirect url to the previous route
+    this.windowService.isXs().pipe(take(1))
```

Comment:
If I understand correctly, we need to determine whether we're on an XS screen, because on XS screens the login component is rendered as a separate page, rather than a dropdown at the top of another page?

If that's the case I'd prefer if you added an input to this component: 
```typescript
@Input() isStandalonePage: boolean
```
or something to that effect, and use _that_ to determine whether or not to set the redirect url.

That way, if we ever change it so that e.g. the page is also used on SM screens, we won't have to update this logic.

Reply:
Yes, that's the idea. Although looking more closely I see that the separate login page is used for both xs and sm screens.  I should have checked that sooner.

I've implemented the @Input, passing "true" from the login-page template and "false" from  auth-nav-menu template. Checked the result and it seems to be working.